cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -167,11 +185,14 @@ public LoadMonitorState state() {
     // Get valid snapshot window number and populate the monitored partition map.
     // We do this primarily because the checker and aggregator are not always synchronized.
     int numValidSnapshotWindows = 0;
-    SortedMap<Long, Double> loadSnapshotsWindows = new TreeMap<>();
+    boolean hasInvalidSnapshotWindows = false;
+    SortedMap<Long, Double> loadSnapshotsWindows = new TreeMap<>(Comparator.reverseOrder());
     for (long window : _metricSampleAggregator.visibleSnapshotWindows()) {
       double monitoredPartitionsPercentage = windowToMonitoredPercentage.getOrDefault(window, 0.0);
-      if (monitoredPartitionsPercentage >= minMonitoredPartitionsPercentage) {
+      if (monitoredPartitionsPercentage >= minMonitoredPartitionsPercentage && !hasInvalidSnapshotWindows) {
```

Comment:
So if there is a window that fell below the minimum number of monitored partitions  then this will clamp the number of valid snapshot windows at the number of contiguous windows starting from the most recent window time index.  Why do you want that?

Reply:
This is the existing behavior. It is primarily because MetricSampleAggregator does not really skip an invalid window and only take valid windows. I agree the behavior is not ideal and we probably want to consider change this. Maybe we can do that in a separate patch.